airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/partial_messages/PartialAirbyteRecordMessage.java

```diff
@@ -15,6 +16,9 @@ public class PartialAirbyteRecordMessage {
   @JsonProperty("stream")
   private String stream;
 
+  @JsonProperty("data")
```

Comment:
> how is this different from a full AirbyteRecordMessage?

Good point, this change now creates parity with `AirbyteRecordMessage`. I can see if the LOE to remove this class is non-trivial, if so I will add to this PR otherwise will file a ticket to remove

> should withSerialized unset the data field so that we can GC the jsonnode?

Can be something to play around with, however, I don't see the value of it right now because memory is not really an issue with 2 Gi on the container

Reply:
neat. I'm curious what problem the partial message was supposed to solve, if we're not seeing any issues with the full message, but that's nonblocking. Agree that if the refactor is messy we should skip it for now, but at least add a comment explaining that we want to remove the partial record class.

Reply:
+1

Reply:
> what problem the partial message was supposed to solve

Initially, it was aimed to be a less memory-intensive deserialized product but the work to pass as much of the message as a serialized string eventually just led to adding in `emitted_at` and `data` which basically led to annotation parity with AirbyteRecordMessage (minus some annotations around `additionalProperties`)

For scope, I'll add a comment first and file out the work that removes this class in the future